StandardSocketsHttpHandler/Net/Http/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -9,10 +9,10 @@ internal static class HttpRequestMessageExtensions
         public static bool HasHeaders(this HttpRequestMessage request)
         {
             // Note: The field name is _headers in .NET core 
-            bool isDotNetFramework = RuntimeUtils.IsDotNetFramework();
-            string headersFieldName = isDotNetFramework ? "headers" : "_headers";
+            bool isDotNetFrameworkOrMono = RuntimeUtils.IsDotNetFramework() || RuntimeUtils.IsMono();
+            string headersFieldName = isDotNetFrameworkOrMono ? "headers" : "_headers";
             FieldInfo headersField = typeof(HttpRequestMessage).GetField(headersFieldName, BindingFlags.Instance | BindingFlags.NonPublic);
-            if (headersField == null && isDotNetFramework)
+            if (headersField == null && isDotNetFrameworkOrMono)

```

Comment:
IMO this should be kept "isDotNetFramework" - no need for the fallback for Mono - we should have gotten it right the first time - right?